test/code/providers/runas_provider/runasprovider_test.cpp

```diff
@@ -517,10 +536,28 @@ class SCXRunAsProviderTest : public CPPUNIT_NS::TestFixture
         CPPUNIT_ASSERT_EQUAL_MESSAGE(ERROR_MESSAGE, L"run-test-unit\n", returnData.stdOut);
         CPPUNIT_ASSERT_EQUAL_MESSAGE(ERROR_MESSAGE, L"-test-\n", returnData.stdErr);
     }
+    
+    void TestDoInvokeMethodScriptTmpDir()
+        {
+            std::wstring errMsg;
+            mi::SCX_OperatingSystem_ExecuteScript_Class param;
+            param.Script_value(
+            		            "#!/bin/sh\n"
+            		            "# Get location pointed by tmpdir\n"
+            					"DIR=`dirname $0`\n"
+            		            "echo $DIR\n"
+            		            "exit 0\n");
+            param.Arguments_value("unit test run");
+            param.timeout_value(0);
+            InvokeReturnData returnData;
+            ExecuteScript(param, MI_RESULT_OK, returnData, CALL_LOCATION(errMsg));
+            CPPUNIT_ASSERT_EQUAL_MESSAGE(ERROR_MESSAGE, 0, returnData.returnCode);
+            CPPUNIT_ASSERT_EQUAL_MESSAGE(ERROR_MESSAGE, L"./tmpdir\n", returnData.stdOut);
+        }
 
     void TestChRoot()
     {
-        system("rm -rf testChRoot > /dev/null 2>&1");
+        int err = system("rm -rf testChRoot > /dev/null 2>&1");
```

Comment:
You likely hit this because you're using some DEV system that's pickier than what we usually use. It's great to clean this stuff up, but it would be nice if you also changed to use SCXProcess::Run() to capture stdout/stderr properly. You should easily see examples of this in other places.